### DISCUSSION TITLE
breeze init failed

### DISCUSSION THREAD

**Discussion by doowhtron** [apache/airflow#18972] [Q&A]
I followed the instructions to initialize breeze environment: https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst#setting-up-breeze

Seems like the image is built but failed to start.
![捕获](https://user-images.githubusercontent.com/4937178/137284517-8870506a-c07d-473c-bc38-d8c30ec4ab42.PNG)

docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/mnt/c/Users/binglilun/source/repos/doowhtron/airflow/scripts/in_container/entrypoint_ci.sh" to rootfs at "/entrypoint" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.

Something wrong with my environment?

**Comment by potiuk** [comment]
As discussed in https://github.com/apache/airflow/issues/18628#issuecomment-943154651 - don't use Windows-mounted folder for Airflow when you are in WSL2. Checkout airflow in a Linux filesystem, not the one mounted from Windows.

**Reply by potiuk** [reply]
And just to add - on top of it did not working well with docker it is also extremely slow to access files mounted from Windows - so you would get a terrible experience anyway https://github.com/microsoft/WSL/issues/4197

**Reply by potiuk** [reply]
See this comment in Breeze WSL2 documentation https://github.com/apache/airflow/blob/main/BREEZE.rst#docker-in-wsl-2:

> WSL 2 Filesystem Performance :
> Accessing the host Windows filesystem incurs a performance penalty, it is therefore recommended to do development on the Linux filesystem. E.g. Run cd ~ and create a development folder in your Linux distro home and git pull the Airflow repo there.

**Reply by doowhtron** [reply]
> As discussed in [#18628 (comment)](https://github.com/apache/airflow/issues/18628#issuecomment-943154651) - don't use Windows-mounted folder for Airflow when you are in WSL2. Checkout airflow in a Linux filesystem, not the one mounted from Windows.

Thanks for your kindness.  @potiuk
I switched the working dir and still got the same error msg. 

```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/home/binglilun/source/doowhtron/airflow/scripts/in_container/entrypoint_ci.sh" to rootfs at "/entrypoint" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.

ERROR: The previous step completed with error. Please take a look at output above

(airflow-env) binglilun@binglilun-NB01:~/source/doowhtron/airflow$ pwd
/home/binglilun/source/doowhtron/airflow
(airflow-env) binglilun@binglilun-NB01:~/source/doowhtron/airflow$
```
I am running docker desktop with the following version and settings. Is it related?
![企业微信截图_16342617831445](https://user-images.githubusercontent.com/4937178/137418704-14c26ded-195b-48cd-b663-907c6107c086.png)

**Reply by potiuk** [reply]
Unfortunately I have no Windows and just leaving for holidays. But if I were you i'd try to experiment with different settings - disabling docker compose 2, enabling the WSL engine instead of Hyper-2. Just paly with it. 

Also i suspect that you could have simply copied the airflow repo checked out on Windows. DON't. Simply check it out from scratch fully on Linux, using Linux filesystem, using Linux git. Copying stuff between windows and Linux might loose some information 


Finally i would (if i were you) simply try before if your docker and doxker-compose work fine with some basic examples - maybe your problems are related to the way you configured your WSL. You can likely google some. Simple getting started with docker/docker compose to see if it works in general.

Curious what you will find out  - this might help others if you find out the reason so kind request to post results of those experiments

**Reply by potiuk** [reply]
Also it REALLY looks like the 'entrypoint_ci.sh' of yours is a directory. I do not know why and how this could happen (but i suspect somehow your checked out airflow sources got messed up so checking out from scratch should help). 

Another thing you can do - Breeze has really nice troubleshooting section - just follow it up and report results of your experiments https://github.com/apache/airflow/blob/main/BREEZE.rst#troubleshooting

**Reply by doowhtron** [reply]
@potiuk The problem is fixed.
Key things for running breeze on my wsl environment are:
1. Upgrade to wsl2(my distribution is ubuntu)
2. checkout the source to ~/ instead of /mnt/c/  
3. run docker daemon inside wsl2(not using docker destop)
4. enable wsl's network by setting /etc/resolv.conf(putting nameserver 8.8.8.8)
5. add "--network host" parameter to docker_v build(_build_images.sh), otherwise internet cannot be connected
6. install yarn and add "--ignore-engines" to yarn install (compile_assets.sh)

And breeze works for me now.

**Reply by potiuk** [reply]
Ah cool. Maybe you could add some docs or even automated adding of network host in wsl2 ? I think it will not work on Mac so just detecting if you areon wsl2 and adding the t might be helpful ?

**Reply by doowhtron** [reply]
OK，I'll try
